content/realtime/connection.textile

```diff
@@ -281,18 +281,33 @@ There are two modes of connection state recovery:
+In @recover@ mode it is necessary to request recovery mode in the "client options":/realtime/usage#client-options when instancing the library. Recovery requires that the library knows the previous connection's <span lang="default">"@recoveryKey@":#recovery-key</span><span lang="ruby">"@recovery_key@":#recovery-key</span> value (which includes both the private unique "@Connection#key@":#key and the last message serial received on that connection). As the recovery key is never shared with any other clients, it allows Ably to safely resend message backlogs to the original client.
 
 blang[javascript].
-  In the browser environment a @window close@ event causes those parameters to be saved in a cookie; a client can then request recovery simply by including @recover: true@ in the initialization options.
+  In the browser environment, if a callback is provided in the @recover@ option, when @window beforeunload@ event fires, the connection details, including the "@recoveryKey@":#recovery-key, are stored in the "browser's sessionStorage":https://developer.mozilla.org/en/docs/Web/API/Window/sessionStorage. The provided @recover@ callback is then invoked whenever the connection state can be recovered and just before a connection is established, passing in the "@LastConnectionDetails@":#last-connection-details. The callback is then responsible for confirming whether the connection state should be recovered or not. For example, it is common to recover connection state when the page is reloaded but not for different pages the user has navigated to. The callback allows the developer to decide if the connection should be recovered or not at the time the new connection is established by inspecting the "@LastConnectionDetails@":#last-connection-details and evaluating that against any other application state. Below is a straightforward example:
 
-  Alternatively, if it is necessary to be explicit about the connection @recoveryKey@ - for example if there are multiple tabs and the cookies would conflict, or if third-party cookies are disabled - the connection can be recovered by providing the last value of the connection's @recoveryKey@ value in the "client options":/realtime/usage#client-options @recover@ attribute when instancing the library.
+  ```[jsall]
+    var ably = new Ably.Realtime({
+      authUrl: '/obtainToken',
+      recover: function(lastConnectionDetails, cb) {
+        if (lastConnectionDetails.url === document.location.href) {
+          cb(true); /* recover connection as user has reloaded page */
+        } else {
+          cb(false); /* do not recover connection as URL has changed */
+        }
+      }
+    });
+  ```
+
+  Alternatively, if it is necessary to be explicit about the connection @recoveryKey@ , the connection can be recovered by providing the last value of the connection's @recoveryKey@ value in the "client options":/realtime/usage#client-options @recover@ attribute when instancing the library.
```

Comment:
> Just wondering if it should be lastConnectionDetails.location or lastConnectionDetails.location.href instead of .url ?

:+1: to stringifying & storing the whole location object, might save some devs the effort of parsing the url. If @mattheworiordan agrees I'll change that in the ably-js pr